tracing-subscriber/src/subscribe/mod.rs

```diff
@@ -1477,6 +1579,120 @@ feature! {
     {
         subscriber_impl_body! {}
     }
+
+
+    impl<C, S> Subscribe<C> for Vec<S>
+    where
+        S: Subscribe<C>,
+        C: Collect,
+    {
+
+        fn on_subscribe(&mut self, collector: &mut C) {
+            for s in self {
+                s.on_subscribe(collector);
+            }
+        }
+
+        fn register_callsite(&self, metadata: &'static Metadata<'static>) -> Interest {
+            // Return highest level of interest.
+            let mut interest = Interest::never();
+            for s in self {
+                let new_interest = s.register_callsite(metadata);
+                if (interest.is_sometimes() && new_interest.is_always())
+                    || (interest.is_never() && !new_interest.is_never())
+                {
+                    interest = new_interest;
+                }
+            }
+
+            interest
+        }
+
+        fn enabled(&self, metadata: &Metadata<'_>, ctx: Context<'_, C>) -> bool {
+            // XXX(eliza): does this need to clear the filter state when it
+            // returns false, or will `Layered` always take care of that for us?
+            self.iter().all(|s| s.enabled(metadata, ctx.clone()))
+        }
+
+        fn on_new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, ctx: Context<'_, C>) {
+            for s in self {
+                s.on_new_span(attrs, id, ctx.clone());
+            }
+        }
+
+        fn max_level_hint(&self) -> Option<LevelFilter> {
+            let mut max_level = LevelFilter::ERROR;
+            for s in self {
+                // NOTE(eliza): this is slightly subtle: if *any* subscriber
+                // returns `None`, we have to return `None`, assuming there is
+                // no max level hint, since that particular subscriber cannot
+                // provide a hint.
+                max_level = core::cmp::max(s.max_level_hint()?, max_level);
```

Comment:
missed the `?` on first read. how about: 
```suggestion
                // NOTE(eliza): this is slightly subtle: if *any* subscriber
                // returns `None`, we have to return `None`, assuming there is
                // no max level hint, since that particular subscriber cannot
                // provide a hint.
                let s = s.max_level_hint()?;
                max_level = core::cmp::max(s, max_level);
```

Reply:
yeah, that seems clearer --- i would not call the subscriber's level hint `s`, though. how about:
missed the `?` on first read. how about: 
```suggestion
                // NOTE(eliza): this is slightly subtle: if *any* subscriber
                // returns `None`, we have to return `None`, assuming there is
                // no max level hint, since that particular subscriber cannot
                // provide a hint.
                let hint = s.max_level_hint()?;
                max_level = core::cmp::max(hint, max_level);
```